### PR TITLE
Move to NetCore 2.0

### LINKED ISSUE #200: Tools on .NET Core 2.0
Will be a breaking change incompatible with .NET Core 1.0 CLI. Requires bumping tools package versions to 2.0.

Alternatively, we can make tasks run in-proc with MSBuild so they are runtime-agnostic. Would still require a 2.0 version bump.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #200.

### CODE REVIEW

File: src/DependenciesPackager/DependenciesPackager.csproj [hunk 1]
```diff
@@ -5,15 +5,14 @@
   <PropertyGroup>
     <Description>Creates a zip package with the expanded version of our nuget packages that can be used as a local cache</Description>
     <VersionPrefix>1.0.1</VersionPrefix>
```
Comment (natemcmaster) [inline src/DependenciesPackager/DependenciesPackager.csproj]: Update this to 2.0.0.

File: build/dependencies.props [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿<Project>
   <PropertyGroup>
+    <AspNetCoreVersion>2.0.0-*</AspNetCoreVersion>
     <CommandLineUtilsVersion>1.2.0-*</CommandLineUtilsVersion>
     <CoreFxVersion>4.3.0</CoreFxVersion>
```
Comment (natemcmaster) [inline build/dependencies.props]: Update to 4.4.0-*

File: src/DependenciesPackager/DependenciesPackager.csproj [hunk 3]
```diff
@@ -5,15 +5,14 @@
...
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" Version="$(CommandLineUtilsVersion)" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="1.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.0.0" />
-    <PackageReference Include="System.Diagnostics.Process" Version="4.1.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="System.Diagnostics.Process" Version="$(CoreFxVersion)" />
```
Comment (natemcmaster) [inline src/DependenciesPackager/DependenciesPackager.csproj]: Not required.

File: src/Internal.AspNetCore.BuildTools.Tasks/Internal.AspNetCore.BuildTools.Tasks.csproj [hunk 4]
```diff
@@ -6,19 +6,10 @@
...
     <DefineConstants>$(DefineConstants);BuildTools</DefineConstants>
     <Description>MSBuild tasks. This package is intended for Microsoft use only</Description>
-
-    <!--
-      The netstandard1.0 TFM doesn't actually compile tasks.
-      It's just there so Internal.AspNetCore.Sdk gets a dependency to this project.
-    -->
-    <TargetFrameworks>netstandard1.6;net46;netstandard1.0</TargetFrameworks>
-    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (natemcmaster) [inline src/Internal.AspNetCore.BuildTools.Tasks/Internal.AspNetCore.BuildTools.Tasks.csproj]: Build tasks still need to cross-compile for .NET Framework. You can't use ns2.0 assemblies from MSBuild (full-framework) because the netstandard shims will not be available.

Also, we still need the netstandard1.0 TFM.

File: src/Internal.AspNetCore.Sdk/Internal.AspNetCore.Sdk.csproj [hunk 5]
```diff
@@ -4,9 +4,8 @@
 
   <PropertyGroup>
     <VersionPrefix>2.1.0</VersionPrefix>
-    <TargetFramework>netstandard1.0</TargetFramework>
```
Comment (natemcmaster) [inline src/Internal.AspNetCore.Sdk/Internal.AspNetCore.Sdk.csproj]: Keep at netstandard1.0. We need this package to be compatible with all TFMs. We still compile some netstandard1.x projects and this would break that.

File: src/Microsoft.AspNetCore.BuildTools.ApiCheck/Microsoft.AspNetCore.BuildTools.ApiCheck.nuspec [hunk 6]
```diff
@@ -6,13 +6,12 @@
     <authors>Microsoft</authors>
     <description>Generates API baselines and detects breaking changes in APIs.</description>
     <dependencies>
-      <group targetFramework=".NETStandard1.0" />
+      <group targetFramework=".NETStandard2.0" />
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.BuildTools.ApiCheck/Microsoft.AspNetCore.BuildTools.ApiCheck.nuspec]: This will prevent us from using API Check on anything that is not netstandard2.0 compatible. Recommend keeping this as ns1.0.

File: src/NuGetPackageVerifier/NuGetPackageVerifier.csproj [hunk 7]
```diff
@@ -24,7 +22,7 @@
     <PackageReference Include="Mono.Cecil" Version="0.10.0-beta1-v2" />
     <PackageReference Include="Newtonsoft.Json" Version="$(JsonNetVersion)" />
     <PackageReference Include="NuGet.Packaging" Version="$(NuGetPackagesVersion)" />
-    <PackageReference Include="System.Runtime.Loader" Version="4.0.0" />
+    <PackageReference Include="System.Runtime.Loader" Version="$(CoreFxVersion)" />
```
Comment (natemcmaster) [inline src/NuGetPackageVerifier/NuGetPackageVerifier.csproj]: Remove this. The package should not be required.

File: src/NuGetPackageVerifier/NuGetPackageVerifier.csproj [hunk 8]
```diff
@@ -7,8 +7,6 @@
     <VersionPrefix>2.0.0</VersionPrefix>
     <TargetFramework>netcoreapp2.0</TargetFramework>
     <AssemblyName>NuGetPackageVerifier</AssemblyName>
-    <OutputType>exe</OutputType>
```
Comment (natemcmaster) [inline src/NuGetPackageVerifier/NuGetPackageVerifier.csproj]: keep this. It's a console app.

File: src/PackageClassifier/PackageClassifier.csproj [hunk 9]
```diff
@@ -5,19 +5,15 @@
   <PropertyGroup>
     <Description>Classifies a set of packages from several directories based on information from a csv file</Description>
     <VersionPrefix>1.0.1</VersionPrefix>
```
Comment (natemcmaster) [inline src/PackageClassifier/PackageClassifier.csproj]: Update this to 2.0.0

File: src/SplitPackages/SplitPackages.csproj [hunk 10]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <Description>Copies NuGet packages form a given source folder into a specific set of folders based on a CSV file.</Description>
     <VersionPrefix>1.0.2</VersionPrefix>
```
Comment (natemcmaster) [inline src/SplitPackages/SplitPackages.csproj]: Update.

File: src/PackagePublisher/PackagePublisher.csproj [hunk 11]
```diff
@@ -5,7 +5,7 @@
     <Description>Publishes packages to a feed, with retries.</Description>
     <DebugType>portable</DebugType>
     <OutputType>exe</OutputType>
-    <TargetFramework>netcoreapp1.0</TargetFramework>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
     <VersionPrefix>1.0.2</VersionPrefix>
```
Comment (natemcmaster) [inline src/PackagePublisher/PackagePublisher.csproj]: Update.

File: src/VersionTool/VersionTool.csproj [hunk 12]
```diff
@@ -3,15 +3,15 @@
...
     <AssemblyName>VersionTool</AssemblyName>
     <OutputType>exe</OutputType>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.DotNet.Cli.Utils" Version="1.0.0-preview2-1-003177" />
+    <PackageReference Include="Microsoft.DotNet.Cli.Utils" Version="$(DotNetCliUtilsVersion)" />
     <PackageReference Include="Microsoft.DotNet.ProjectModel" Version="$(DotNetProjectModelVersion)" />
-    <PackageReference Include="Microsoft.Extensions.CommandLineUtils" Version="1.0.0" />
+    <PackageReference Include="Microsoft.Extensions.CommandLineUtils" Version="$(CommandLineUtilsVersion)" />
```
Comment (natemcmaster) [inline src/VersionTool/VersionTool.csproj]: Change to Microsoft.Extensions.CommandLineUtils.Sources and use 2.0.0-\*

File: test/ApiCheck.Test/ApiCheck.Test.csproj [hunk 13]
```diff
@@ -3,8 +3,7 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp2.0</TargetFrameworks>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (natemcmaster) [inline test/ApiCheck.Test/ApiCheck.Test.csproj]: Keep the .NET Framework tests. It's important that we ensure this tool still runs on .NET Framework.

File: src/NugetReferenceResolver/NugetReferenceResolver.csproj [hunk 14]
```diff
@@ -5,13 +5,13 @@
   <PropertyGroup>
     <Description>Resolves the dependency graph for a given set of project dependencies and computes different subsets of the dependency graph</Description>
     <VersionPrefix>1.0.1</VersionPrefix>
-    <TargetFrameworks>netstandard1.6;net451</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <OutputType>library</OutputType>
```
Comment (natemcmaster) [inline src/NugetReferenceResolver/NugetReferenceResolver.csproj]: Nit: as long as we're doing cleanup, we can remove this as it is the default value. This is true in a few other projects as well.

File: build/dependencies.props (hunk 2 above)
Comment (ryanbrandenburg) [inline build/dependencies.props]: The only package using this is `System.Runtime.Serialization.Primitives` and its latest version on NuGet.org is 4.3.0, it doesn't appear on our feeds. Maybe this property just needs a more accurate name?

File: src/PackageClassifier/PackageClassifier.csproj [hunk 15]
```diff
@@ -4,20 +4,15 @@
 
   <PropertyGroup>
     <Description>Classifies a set of packages from several directories based on information from a csv file</Description>
-    <VersionPrefix>1.0.1</VersionPrefix>
-    <TargetFrameworks>netstandard1.6;net451</TargetFrameworks>
-    <OutputType>library</OutputType>
+    <VersionPrefix>2.0.0</VersionPrefix>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
```
Comment (ryanbrandenburg) [inline src/PackageClassifier/PackageClassifier.csproj]: I had to bump this to netstandard2.0 in order to take the logging packages.

**Comment by ryanbrandenburg** [comment]
:up::date:

File: src/VersionTool/VersionTool.csproj [hunk 16]
```diff
@@ -3,15 +3,15 @@
...
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp1.0</TargetFramework>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
     <AssemblyName>VersionTool</AssemblyName>
     <OutputType>exe</OutputType>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.DotNet.Cli.Utils" Version="1.0.0-preview2-1-003177" />
```
Comment (natemcmaster) [inline src/VersionTool/VersionTool.csproj]: The reason this was pinned to an old version of the package is that it is a project.json tool. I'm not sure this will work if you lift it to the 1.0.1 version of this package. It may be better to leave this one untouched until we completely fix it to work with csproj. See https://github.com/aspnet/BuildTools/issues/270 

cc @JunTaoLuo

File: src/Internal.AspNetCore.BuildTools.Tasks/GetOSPlatform.cs [hunk 17]
```diff
@@ -22,10 +20,6 @@ public class GetOSPlatform : Task
 
         public override bool Execute()
         {
-#if NET46
```
Comment (natemcmaster) [inline src/Internal.AspNetCore.BuildTools.Tasks/GetOSPlatform.cs]: Undo this change too.

File: build/dependencies.props [hunk 18]
```diff
@@ -1,6 +1,7 @@
 ﻿<Project>
   <PropertyGroup>
-    <CommandLineUtilsVersion>1.2.0-*</CommandLineUtilsVersion>
+    <AspNetCoreVersion>2.0.0-*</AspNetCoreVersion>
+    <CommandLineUtilsVersion>2.0.0-*</CommandLineUtilsVersion>
```
Comment (natemcmaster) [inline build/dependencies.props]: Nit: this can be the same as AspNetCoreVersion as the M.Ext.CommandLineUtils.Sources package revs with the rest of aspnet.

File: src/SplitPackages/SplitPackages.csproj [hunk 19]
```diff
@@ -15,8 +15,8 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" Version="$(CommandLineUtilsVersion)" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="1.0.0-*" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.0.0-*" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="$(AspNetCoreVersion)" />
     <PackageReference Include="Newtonsoft.Json" Version="$(JsonNetVersion)" />
     <PackageReference Include="NuGet.Client" Version="$(NuGetPackagesVersion)" />
     <PackageReference Include="System.Runtime.Serialization.Primitives" Version="$(CoreFxVersion)" />
```
Comment (natemcmaster) [inline src/SplitPackages/SplitPackages.csproj]: This package reference should be removed. It's unnecessary in netcoreapp2.0 projects.

File: src/VersionTool/VersionTool.csproj (hunk 16 above)
Comment (JunTaoLuo) [inline src/VersionTool/VersionTool.csproj]: Yup, let's leave this one.

**Comment by ryanbrandenburg** [comment]
:up::date:

File: src/Internal.AspNetCore.BuildTools.Tasks/Internal.AspNetCore.BuildTools.Tasks.csproj [hunk 20]
```diff
@@ -3,7 +3,7 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <VersionPrefix>1.0.0</VersionPrefix>
+    <VersionPrefix>2.0.0</VersionPrefix>
```
Comment (natemcmaster) [inline src/Internal.AspNetCore.BuildTools.Tasks/Internal.AspNetCore.BuildTools.Tasks.csproj]: Since it appears we've now undone all changes to this project, you can keep the version at 1.0.0.

**Comment by ryanbrandenburg** [comment]
Don't want to merge this today in case it interfeers with the build. Will wait until Monday when the build failing isn't critical.

**Comment by ryanbrandenburg** [comment]
Repos which need a reaction:
- [ ] Coherence-Signed
- [ ] AspNetCoreModule
- [ ] Universe
- [ ] CoreCLR
